index.js

```diff
@@ -211,13 +213,21 @@ Sharp.prototype.flatten = function(flatten) {
 };
 
 Sharp.prototype.overlayWith = function(overlayPath) {
-  if (typeof overlayPath !== 'string') {
-    throw new Error('The overlay path must be a string');
-  }
-  if (overlayPath === '') {
-    throw new Error('The overlay path cannot be empty');
+  try {
+    var colour = color(overlayPath);
+    this.options.overlayColor = colour.rgbArray();
+    this.options.overlayColor.push(colour.alpha() * 255);
```

Comment:
Thanks Lovell, I will get these fixed up in the next commit :)

Sent using CloudMagic [https://cloudmagic.com/k/d/mailapp?ct=pi&cv=6.0.64&pv=8.2]
On Fri, Sep 25, 2015 at 23:05, Lovell Fuller <notifications@github.com> wrote:
In index.js [https://github.com/lovell/sharp/pull/272#discussion_r40439276] :

> @@ -211,13 +213,21 @@ Sharp.prototype.flatten = function(flatten) {
>  };
>  
>  Sharp.prototype.overlayWith = function(overlayPath) {
> -  if (typeof overlayPath !== 'string') {
> -    throw new Error('The overlay path must be a string');
> -  }
> -  if (overlayPath === '') {
> -    throw new Error('The overlay path cannot be empty');
> +  try {
> +    var colour = color(overlayPath);
> +    this.options.overlayColor = colour.rgbArray();
> +    this.options.overlayColor.push(colour.alpha() * 255);


We might want to Math.round the alpha value as it gets clipped to an integer in
C++ land. (I've spotted that the existing background method should do this also.)

—
Reply to this email directly or view it on GitHub [https://github.com/lovell/sharp/pull/272/files#r40439276] .[https://github.com/notifications/beacon/AAGXzfE2ivRT5o6xjzSLgHr70TBnJvNNks5o1VozgaJpZM4GAZoR.gif]